tests/test_pipelines_object_detection.py

```diff
@@ -0,0 +1,263 @@
+
+        object_detector = pipeline("object-detection", model=model_id)
+
+        outputs = object_detector("http://images.cocodataset.org/val2017/000000039769.jpg")
+
+        self.assertGreater(len(outputs), 0)
+        for detected_object in outputs:
+            self.assertEqual(
+                detected_object,
+                {
+                    "score": ANY(float),
+                    "label": ANY(str),
+                    "box": {"xmin": ANY(int), "ymin": ANY(int), "xmax": ANY(int), "ymax": ANY(int)},
+                },
+            )
+
+        batch = [
+            "http://images.cocodataset.org/val2017/000000039769.jpg",
+            "http://images.cocodataset.org/val2017/000000039769.jpg",
+        ]
+        batch_outputs = object_detector(batch)
+
+        self.assertEqual(len(batch), len(batch_outputs))
+        for outputs in batch_outputs:
+            self.assertGreater(len(outputs), 0)
+            for detected_object in outputs:
+                self.assertEqual(
+                    detected_object,
+                    {
+                        "score": ANY(float),
+                        "label": ANY(str),
+                        "box": {"xmin": ANY(int), "ymin": ANY(int), "xmax": ANY(int), "ymax": ANY(int)},
+                    },
+                )
+
+    def test_custom_tokenizer(self):
+        model_id = "facebook/detr-resnet-50"
+        tokenizer = PreTrainedTokenizer()
+
+        # Assert that the pipeline can be initialized with a feature extractor that is not in any mapping
+        object_detector = pipeline("object-detection", model=model_id, tokenizer=tokenizer)
+
+        self.assertIs(object_detector.tokenizer, tokenizer)
+
+    def test_low_threshold(self):
+        threshold = 0.0
+        model_id = "mishig/tiny-detr-mobilenetsv3"
+
+        model = AutoModelForObjectDetection.from_pretrained(model_id)
+        feature_extractor = AutoFeatureExtractor.from_pretrained(model_id)
+        object_detector = ObjectDetectionPipeline(model=model, feature_extractor=feature_extractor)
+
+        valid_input = {"images": "http://images.cocodataset.org/val2017/000000039769.jpg", "threshold": threshold}
+
+        output = object_detector(**valid_input)
+
+        self.assertTrue(isinstance(output, list))
+        self.assertEqual(len(output), 5)
+
+    def test_high_threshold(self):
+        threshold = 1.0
+        model_id = "mishig/tiny-detr-mobilenetsv3"
+
+        model = AutoModelForObjectDetection.from_pretrained(model_id)
+        feature_extractor = AutoFeatureExtractor.from_pretrained(model_id)
+        object_detector = ObjectDetectionPipeline(model=model, feature_extractor=feature_extractor)
+
+        valid_input = {"images": "http://images.cocodataset.org/val2017/000000039769.jpg", "threshold": threshold}
+
+        output = object_detector(**valid_input)
+
+        self.assertTrue(isinstance(output, list))
+        self.assertEqual(len(output), 0)
```

Comment:
I think we can fuse `low_threshold` and `high_threshold` into 1 test, which would be testing a bit more specifically.

Currently the test would be failing on a lot of changes in the threshold code (you're only testing the extreme cases).

I think a better test would be to first get all top_k = 5 first results, take a threshold between 2nd and 3rd options (supposing they are different).

Because this test is deterministic we can even specify a hardcoded value for that threshold.
IMO that would be a better in checking that the filtering is filtering exactly what we expect it to.